src/node/HttpBodyImpl.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpBody } from "../http";
+import { stringToReadableStream } from "./utils";
+
+export class HttpBodyImpl implements HttpBody {
+  constructor(public readonly stream: ReadableStream) {}
+
+  public static fromString(content: string): HttpBody {
+    return new HttpBodyImpl(stringToReadableStream(content));
+  }
+
+  async asJson<T>(): Promise<T> {
+    const bodyToString = await this.asString();
+
+    return JSON.parse(bodyToString);
+  }
+
+  async asString(encoding: string = "utf8") {
+    const chunks: any[] = [];
+    const reader = this.stream.getReader();
+
+    return new Promise<string>(async (resolve, reject) => {
+      while (true) {
+        try {
+          const { done, value } = await reader.read();
+          if (done) {
+            resolve(Buffer.concat(chunks).toString(encoding));
```

Comment:
I think we have to remove `Buffer` here.

Reply:
The incoming request from Node is a stream of byte array aka a Buffer. I don't know any better way of concatenating it.

Reply:
We should be able to move this class to http4ts and make it environment independent. That'w why I say `Buffer` should not be here. I found this working sample that converts `StreamReader` results to `string`:

``` js
const decoder = new TextDecoder("utf-8");

(async () => {
  const fetchedResource = await fetch("https://flaviocopes.com/");
  const reader = fetchedResource.body.getReader();

  let result = "";

  reader.read().then(function processText({ done, value }) {
    if (done) {
      console.log("Stream finished. Content received:");
      console.log(result);
      return;
    }

    console.log(`Received ${result.length} chars so far!`);

    result += decoder.decode(value);

    return reader.read().then(processText);
  });
})();
```

Reply:
I have created #42 to followup on this.